contracts/lbp/LBPWrapper.sol

```diff
@@ -167,6 +167,69 @@ contract LBPWrapper {
         vault.joinPool(lbp.getPoolId(), address(this), address(this), request);
     }
 
+    /**
+     * @dev exit pool or remove liquidity from pool
+     * @param _tokens                   array of tokens sorted for the LBP
+     * @param _receiver                 address who will be credited with the tokens after removing liquidity
+     * @param _minAmountsOut            array of minimum amounts out
+     * @param _toInternalBalance        fund tokens to the internal user balance
+     * @param _userData                 userData specifies the type of exit
+     */
+    function removeLiquidity(
+        IERC20[] memory _tokens,
+        address payable _receiver,
+        uint256[] memory _minAmountsOut,
+        bool _toInternalBalance,
+        bytes memory _userData
+    ) external {
+        require(
+            _receiver != payable(address(0)),
+            "LBPWrapper: receiver of project and funding tokens can't be zero"
+        );
+        require(holdsPoolTokens(), "LBPWrapper: pool tokens were withdrawn");
+        require(endTimeReached(), "LBPWrapper: cannot exit before endtime");
+
+        IVault vault = lbp.getVault();
+
+        IVault.ExitPoolRequest memory request = IVault.ExitPoolRequest({
+            minAmountsOut: _minAmountsOut,
+            userData: _userData,
+            toInternalBalance: _toInternalBalance,
+            assets: _tokens
+        });
+
+        lbp.approve(address(vault), lbp.balanceOf(address(this)));
```

Comment:
Why are we approving the vault again for the balance? Do we need to do this because the balance in the lbp has changed because of the fundraiser? And if so, don't we have to approve both tokens again?

Reply:
here, we are approving the vault, for the pool tokens.

Reply:
Ahh okay. So we approve the Vault for the tokens, and when transfeering them, there's a sort of swap happening? Just for my understanding

Reply:
kind of, pool tokens swapped for project and funding tokens. Here pool tokens are burned by LBP.